cosmoz-default-tree.html

```diff
@@ -15,142 +27,236 @@
+			nodes = [];
 
-		for (i = 0; i < this._roots.length; i+=1) {
-			node = this._roots[i];
-			if (node.pathLocator === pathLocator) {
-				return node;
+		// Get the nodes on the path.
+		path.some(function (nodeKey) {
+			var node = pathSegment[nodeKey],
+				children = node ? node[this.childProperty] : undefined;
+			
+			if (node) {
+				nodes.push(node);
 			}
 
-			if (pathLocator.indexOf(node.pathLocator + '.') === 0) {
-				path = pathLocator.slice(node.pathLocator.length + 1).split('.');
-				for (i = 0; i < path.length; i+=1) {
-					if (node.children) {
-						node = node.children[path[i]];
-					} else {
-						node = null;
-					}
-					if (node === undefined ||  node === null) {
-						break;
-					}
-				}
-
-				if (node !== undefined && node !== null) {
-					return node;
-				}
+			if (node && children !== undefined && Object.keys(children).length > 0) {
+				pathSegment = children;
+			} else {
+				return true;
 			}
+		}, this);
+
+		return nodes;
+	},
+
+	/**
+	 * Returns a string which describes the path of a node (found by its path locator).
+	 * @return {string} e.g. home/computer/desktop
+	 * @param {string} pathLocator (The string which describes the path. e.g. "1.2.9")
+	 * @param {string} pathProperty (The property of a node on which the path should be build on. e.g "location" with node = {"location": "home", ..})
+	 * @param {string} pathSeparator ["/"] (The string the path should get separated with.)
+	 * @param {string} separatorSign ["."] (The string which separates the path segments of pathLocator.)
+	*/
+	Cosmoz.DefaultTree.prototype.getPathString = function (pathLocator, pathProperty, pathSeparator, separatorSign) {
+		var pathNodes = this.getPathNodes(pathLocator, this._treeData);
+
+		if (!pathNodes) {
+			return;
 		}
+
+		// Defaults
+		pathSeparator = pathSeparator || '/';
+		separatorSign = separatorSign || '.';
+
+		return pathNodes.map(function (node) {
+			return node[pathProperty];
+		}).join(pathSeparator);
 	};
 
-	Cosmoz.DefaultTree.prototype._getPathByPathLocator = function (pathLocator) {
-		var	node,
-			path,
-			subPath,
-			i;
+	/**
+	 * Returns a string which describes the path of a node (found by a node's property and value).
+	 * @return {string} e.g. home/computer/desktop
+	 * @param {string} propertyName (The name of the property the match should be based on. e.g. "name")
+	 * @param {string} propertyValue (The value of the property the match should be based on. e.g. "Peter")
+	 * @param {string} pathProperty (The property of a node on which the path should be build on. e.g "location" if node = {"location": "home"})
+	 * @param {string} pathSeparator (The string the path should get separated with.) Default: "/"
+	 * @param {string} separatorSign (The string which separates the path. e.g ".") Default: "."
+	*/
+	Cosmoz.DefaultTree.prototype.getPathStringByProperty = function (propertyName, propertyValue, pathProperty, pathSeparator, separatorSign) {
```

Comment:
What's going on with `propertyName` + `pathProperty`?
How are they used?
For me, they feel like the same, and if they are, they/it should default to `this.searchProperty`, not hard-coded `'name'`.
Also, since they do have defaults, they should be moved after required values like `propertyValue` to make them optional.